packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -191,6 +197,24 @@ export class HardhatNode extends EventEmitter {
           }
         }
       }
+
+      if (config.chains.has(forkNetworkId)) {
+        hardforkActivations = config.chains.get(forkNetworkId)!.hardforkHistory;
+      } else if (defaultHardhatNetworkParams.chains.has(forkNetworkId)) {
+        hardforkActivations =
+          defaultHardhatNetworkParams.chains.get(
+            forkNetworkId
+          )!.hardforkHistory;
+      } else {
+        throw new InternalError(
+          `No hardfork history configured for chain ID ${forkNetworkId}`
+        );
+      }
+      hardforkActivations.forEach((block, hardforkName) => {
+        if (!HARDHAT_NETWORK_SUPPORTED_HARDFORKS.includes(hardforkName)) {
+          throw new InternalError(`Unsupported hardfork ${hardforkName}`);
```

Comment:
Same for this error message. It should give a better explanation.

Reply:
This particular `throw` is gone now, but you can see the improved error messaging in [this commit](https://github.com/nomiclabs/hardhat/pull/1928/commits/f6a5dbe1fe1aa1007192aa4e2cf31e1f1c7c5426).